### DISCUSSION TITLE
Private Nexus repository usage for dependencies upgrade

### DISCUSSION THREAD

**Discussion by dioniseo** [renovatebot/renovate#9255] [Request Help] [answered]
Hello, we are using private nexus repositories for pulling all dependencies for our services. This Nexus repository is configured in build.gradle files in repositories section.

Am I right that if this private repository is configured in build.gradle script then Renovate Bot will automatically start looking for newer versions of dependencies in this Nexus registry and not in public ones? Or some additional configurations are needed in config file of Renovate Bot? If so are there any documentation about this?

Unfortunately didn't find much about configuration of private repos except some `hostRules` and `packageRules` configs like this:
```
"packageRules": [
    {
      "managers": ["gradle"],
      "registryUrls": [
        "https://my.nexus.repo/xxx/zzz/",
        "https://other.nexus/"
      ]
    }
  ],
```
```
hostRules: [{
      "hostName": "my.nexus.repo",
      "username": "xxxxx",
      "password": "yyyyy"
  }],
```
Is it correct way of configuring private Nexus registries for new dependencies versions lookup instead of public ones?

Thanks!

**Comment by rarkins** [comment, marked as answer]
Yes, that's the correct way to configure a private registry. Sometimes the manual package rule may not be necessary if those URLs are already specified in the package file and discovered by Renovate, but it can't hurt to provide it anyway.

**Reply by dioniseo** [reply]
Ok, thanks for confirming it!
